### PR TITLE
Provide event name in `runtime.Notify`

### LINKED ISSUE #1052: Change Runtime.Notify to accept explicit event name
Follow a nice change of neo-project/neo#1701.

### PR DESCRIPTION
**Pull request by fyrchik**
Closes #1052 .
Draft because for `v0.91.0`.

### CODE REVIEW

File: pkg/core/interop_system.go [hunk 1]
```diff
@@ -216,19 +216,23 @@ func runtimeGetTrigger(ic *interop.Context, v *vm.VM) error {
 // runtimeNotify should pass stack item to the notify plugin to handle it, but
 // in neo-go the only meaningful thing to do here is to log.
 func runtimeNotify(ic *interop.Context, v *vm.VM) error {
-	// It can be just about anything.
-	e := v.Estack().Pop()
-	item := e.Item()
+	name := v.Estack().Pop().Bytes()
+	elem := v.Estack().Pop()
+	args := elem.Array()
```
Comment (roman-khimov) [inline pkg/core/interop_system.go:221]: Why do you force it being an array? It can be anything.

File: pkg/core/interop_system.go [hunk 2]
```diff
@@ -216,19 +216,23 @@ func runtimeGetTrigger(ic *interop.Context, v *vm.VM) error {
 // runtimeNotify should pass stack item to the notify plugin to handle it, but
 // in neo-go the only meaningful thing to do here is to log.
 func runtimeNotify(ic *interop.Context, v *vm.VM) error {
-	// It can be just about anything.
-	e := v.Estack().Pop()
-	item := e.Item()
+	name := v.Estack().Pop().Bytes()
```
Comment (roman-khimov) [inline pkg/core/interop_system.go:219]: It's limited with `MaxEventName`.

File: pkg/core/state/notification_event.go [hunk 3]
```diff
@@ -12,7 +14,8 @@ import (
 // notification and that item itself.
 type NotificationEvent struct {
 	ScriptHash util.Uint160
-	Item       stackitem.Item
+	Name       string
+	Item       *stackitem.Array
```
Comment (roman-khimov) [inline pkg/core/state/notification_event.go:18]: It still is `stackitem.Item`.

File: pkg/compiler/vm_test.go [hunk 4]
```diff
@@ -21,7 +22,14 @@ type testCase struct {
 
 func runTestCases(t *testing.T, tcases []testCase) {
 	for _, tcase := range tcases {
-		t.Run(tcase.name, func(t *testing.T) { eval(t, tcase.src, tcase.result) })
+		if tcase.name != "struct methods with arguments" {
+			continue
+		}
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/compiler/vm_test.go:27]: Fixed.

File: pkg/core/interop_system.go (hunk 1 above)
Comment (fyrchik) [reply pkg/core/interop_system.go:249]: Not anymore according to https://github.com/neo-project/neo/blob/master/src/neo/SmartContract/ApplicationEngine.Runtime.cs#L150

File: pkg/core/interop_system.go (hunk 2 above)
Comment (fyrchik) [reply pkg/core/interop_system.go:249]: Fixed.

**Pull request merged**